src/providers/authProvider/index.ts

```diff
@@ -72,10 +72,11 @@ const authProvider = (dataProvider: DataProvider): AuthProvider => {
       }
     },
     logout: async (): Promise<void> => {
+      const user = getUser()
       await audit({
         activityType: AuditType.LOGOUT,
-        resource: null,
-        dataId: null,
+        resource: constants.R_USERS,
+        dataId: user?.id ?? null,
```

Comment:
@TahaKhanAbdalli - do you understand the occasions when `user` could be `null`?   I _presume_ we only offer `logout` for logged-in users.

Reply:
Yes. If the user is not logged in then `dataId` would be `null` but if the user is `loggedIn` the user returned from `getUser` has a type of undefined or User, therefore, `user?.id`